### PR TITLE
RUN-3358 remote subscription loop

### PR DESCRIPTION
**Pull request by rdepena**
We were missing the runtimeUuid stamp on the events raised in a multi-runtime environment.  Additionally we had scenarios where securityRealm was not taken into consideration while matching subscriptions.

### Tests:
[Win10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59c959b07cb79f732d10c3a9)
[Win7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59c959b07cb79f732d10c3a9)

### CODE REVIEW

File: src/browser/connection_manager.ts [hunk 1]
```diff
@@ -12,15 +12,15 @@ Please contact OpenFin Inc. at sales@openfin.co to obtain a Commercial License.
...
 
 // npm modules
 // (none)
 
 // local modules
-const coreState = require('./core_state');
+import { Identity } from './api_protocol/transport_strategy/api_transport_base';
+import { ArgMap, PortInfo } from './port_discovery';
 import * as log from './log';
+const coreState = require('./core_state');
```
Comment (whyn07m3) [inline src/browser/connection_manager.ts]: `import` here?

File: src/browser/api_protocol/transport_strategy/ws_strategy.ts [hunk 2]
```diff
@@ -49,6 +50,11 @@ export class WebSocketStrategy extends ApiTransportBase<MessagePackage> {
     }
 
     public send(externalConnection: any, payload: any): void {
+        try {
+            log.writeToLog('info', `sent external-adapter <= ${externalConnection.id} ${JSON.stringify(payload)}`);
+        } catch (err) {
+            /* tslint:disable: no-empty */
+        }
         socketServer.send(externalConnection.id, JSON.stringify(payload));
```
Comment (whyn07m3) [inline src/browser/api_protocol/transport_strategy/ws_strategy.ts]: `JSON.stringify(payload)` can blow up here as well, just like in the logging above...

File: src/browser/connection_manager.ts [hunk 3]
```diff
@@ -73,10 +73,19 @@ function isMeshEnabled(args: ArgMap) {
     return enabled;
 }
 
+function keyFromPortInfo(portInfo: PortInfo): string {
+    const { version, port, securityRealm } = portInfo;
+    return `${version}/${port}/${securityRealm ? securityRealm : ''}`;
```
Comment (whyn07m3) [inline src/browser/connection_manager.ts]: `const { version, port, securityRealm = '' } = portInfo;` ?

File: src/browser/connection_manager.ts [hunk 4]
```diff
@@ -73,10 +73,19 @@ function isMeshEnabled(args: ArgMap) {
...
+}
+
 if (multiRuntimeEnabled && isMeshEnabled(coreState.argo)) {
     startConnectionManager();
 }
 
+function getMeshUuid(): string {
+    const portInfo = <PortInfo>coreState.getSocketServerState();
```
Comment (whyn07m3) [inline src/browser/connection_manager.ts]: I might be wrong, but shouldn't it be `const portInfo: PortInfo = coreState.getSocketServerState();` or it's the same thing just different signature...?

Comment (rdepena) [reply src/browser/connection_manager.ts]: getSocketServer state is not typed.

File: src/browser/api_protocol/transport_strategy/ws_strategy.ts (hunk 2 above)
Comment (rdepena) [reply src/browser/api_protocol/transport_strategy/ws_strategy.ts]: Thats ok, that is how it used to be.

**Pull request merged**